index.js

```diff
@@ -39,30 +40,100 @@ function getTorrentMetaBuffer () { // TODO Fix parsing bug so this can be remove
 }
 
 function getFile (fpath) {
-  return _getFile(this, fpath)
+  return _getFile(module.exports, fpath)
+  .then(file => {
+    debug('FILE path=' + (file || {}).path, 'found=' + (file != null), file)
+    return file
+  })
+}
+
+function fetch (req, opts) {
+  opts = opts || {}
+  let inject = false
+  let url = null
+
+  // Convert a FetchEvent to Request
+  if (global.FetchEvent && req instanceof global.FetchEvent) {
+    inject = req.clientId == null
+    req = req.request // req is now an instance of Request
+  }
+
+  // Convert a Request to an URL
+  if (req instanceof global.Request) {
+    url = new URL(req.url)
+    inject = inject && url.search.substr(1)
+             .split('&').find(s => s === 'noPlanktosInjection') == null
+    if (req.method !== 'GET') throw new Error('Only HTTP GET requests supported')
+  } else if (req instanceof URL) {
+    url = req
+  } else if (typeof req === 'string') {
+    url = new URL(req)
+  }
+
+  if (req == null) throw new Error('Must provide a FetchEvent, Request, URL, or a string')
+  if (url.origin !== global.location.origin) throw new Error('Cannot Fetch. Origin differs')
+
+  inject = 'inject' in opts ? opts.inject : inject
+
+  debug('FETCH-REQ url=' + url.pathname, 'inject=' + inject)
+
+  // Generate response blob. Depends on if the downloader should be injected or not
+  let blobPromise = null
+  if (inject) {
+    let fname = url.pathname.substr(url.pathname.lastIndexOf('/') + 1)
+    const isHTML = fname.endsWith('.html') || fname.endsWith('.htm') || !fname.includes('.')
+    let modUrl = new URL(url.toString())
+    modUrl.search = (url.search === '' ? '?' : url.search + '&') + 'noPlanktosInjection'
+    let html = (isHTML ? injection.docWrite : injection.iframe)
+               .replace('{{url}}', modUrl.toString())
+               .replace('{{root}}', opts.root ? opts.root : '')
+    blobPromise = Promise.resolve(new Blob([html], {type: 'text/html'}))
+  } else {
+    // fpath is relative to the service worker scope if opts.root was given
+    let fpath = opts.root ? url.pathname.replace(opts.root, '') : url.pathname
+
+    blobPromise = Promise.all([
+      global.caches.open('planktos')
+        .then(c => c.match(path.normalize(url.pathname)))
+        .then(r => r ? r.blob() : undefined),
```

Comment:
Would helpt to rename `r` to `resp`. I couldn't disambiguate this without referencing the docs